src/aws/lambda-encode/index.spec.js

```diff
@@ -0,0 +1,206 @@
+    const event = {
+      Records: [
+        {
+          s3: {
+            bucket: {
+              name: 'source bucket',
+            },
+            object: {
+              key:
+                '630dfaaa-8b1c-4d2e-b708-c9a2d715cf59/timedtexttrack/dba1512e-d0b3-40cc-ae44-722fbe8cba6a/1542967735_fr',
+            },
+          },
+        },
+      ],
+    };
+
+    it('delegates to encodeTimedTextTrack and calls updateState when it succeeds', async () => {
+      await lambda(event, null, callback);
+
+      expect(mockEncodeTimedTextTrack).toHaveBeenCalledWith(
+        '630dfaaa-8b1c-4d2e-b708-c9a2d715cf59/timedtexttrack/dba1512e-d0b3-40cc-ae44-722fbe8cba6a/1542967735_fr',
+        'source bucket',
+      );
+      expect(mockUpdateState).toHaveBeenCalledWith(
+        '630dfaaa-8b1c-4d2e-b708-c9a2d715cf59/timedtexttrack/dba1512e-d0b3-40cc-ae44-722fbe8cba6a/1542967735_fr',
+        'ready',
+      );
+    });
+
+    it('delegates to encodeTimedTextTrack and reports the error when it fails', async () => {
+      mockEncodeTimedTextTrack.mockImplementation(
+        () => new Promise((resolve, reject) => reject('Failed!')),
+      );
+
+      await lambda(event, null, callback);
+
+      expect(mockEncodeTimedTextTrack).toHaveBeenCalledWith(
+        '630dfaaa-8b1c-4d2e-b708-c9a2d715cf59/timedtexttrack/dba1512e-d0b3-40cc-ae44-722fbe8cba6a/1542967735_fr',
+        'source bucket',
+      );
+      expect(callback).toHaveBeenCalledWith('Failed!');
+    });
+  });
+
+  describe('called with a video object', () => {
+    const event = {
+      Records: [
+        {
+          s3: {
+            bucket: {
+              name: 'source bucket',
+            },
+            object: {
+              key:
+                '630dfaaa-8b1c-4d2e-b708-c9a2d715cf59/video/dba1512e-d0b3-40cc-ae44-722fbe8cba6a/1542967735_fr',
+            },
+          },
+        },
+      ],
+    };
+
+    it('delegates to encodeVideo and calls updateState & callback when it succeeds', async () => {
+      await lambda(event, null, callback);
+
+      expect(mockEncodeVideo).toHaveBeenCalledWith(
+        '630dfaaa-8b1c-4d2e-b708-c9a2d715cf59/video/dba1512e-d0b3-40cc-ae44-722fbe8cba6a/1542967735_fr',
+        'source bucket',
+      );
+      expect(mockUpdateState).toHaveBeenCalledWith(
+        '630dfaaa-8b1c-4d2e-b708-c9a2d715cf59/video/dba1512e-d0b3-40cc-ae44-722fbe8cba6a/1542967735_fr',
+        'processing',
+      );
+    });
+
+    it('delegates to encodeVideo and reports the error when it fails', async () => {
+      mockEncodeVideo.mockImplementation(
+        () => new Promise((resolve, reject) => reject('Failed!')),
+      );
+
+      await lambda(event, null, callback);
+
+      expect(mockEncodeVideo).toHaveBeenCalledWith(
+        '630dfaaa-8b1c-4d2e-b708-c9a2d715cf59/video/dba1512e-d0b3-40cc-ae44-722fbe8cba6a/1542967735_fr',
+        'source bucket',
+      );
+      expect(callback).toHaveBeenCalledWith('Failed!');
+    });
+  });
+});
```

Comment:
in the above failure cases, maybe you should also check that mockUpdateState was not called.